src/main/java/duckie/Parser.java

```diff
@@ -0,0 +1,135 @@
+package duckie;
+
+import java.time.LocalDate;
+import java.time.LocalDateTime;
+import java.time.format.DateTimeFormatter;
+
+import duckie.command.AddCommand;
+import duckie.command.ByeCommand;
+import duckie.command.Command;
+import duckie.command.DeleteAllCommand;
+import duckie.command.DeleteCommand;
+import duckie.command.DoneCommand;
+import duckie.command.FindCommand;
+import duckie.command.ListCommand;
+import duckie.exception.DuckieException;
+import duckie.exception.DuckieInsufficientInfoException;
+import duckie.exception.DuckieInvalidCommandException;
+import duckie.task.Deadline;
+import duckie.task.Event;
+import duckie.task.Task;
+import duckie.task.Todo;
+
+/**
+ * Responsible for the parsing of input Commands
+ */
+public class Parser {
+    private static boolean isAWord(String s) {
+        return (s.length() > 0 && s.split("\\s+").length == 1);
+    }
+
+    /**
+     * Parse the input Command to direct Duckie on what actions to carry out
+     * @param fullCommand Input string command
+     * @return Specific Command
+     * @throws DuckieException
+     */
+    public static Command parse(String fullCommand) throws DuckieException {
+        String input = fullCommand.strip();
+        if (input.equalsIgnoreCase("bye")) {
+            return new ByeCommand();
+        } else if (input.equalsIgnoreCase("list")) {
+            return new ListCommand();
+        } else if (input.toLowerCase().indexOf("done") == 0) {
+            if (isAWord(input)) {
+                throw new DuckieInsufficientInfoException();
+            }
+
+            try {
+                Integer.parseInt(input.split(" ")[1]);
+            } catch (NumberFormatException e) {
+                throw new DuckieException("You have to input a number after 'done'!");
```

Comment:
Could consider making a custom exception here instead of throwing a DuckieException.